### PR TITLE
AP-224 Provider capital 1d. Shared ownership

### PR DESCRIPTION
**Pull request by tmckernan**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/AP-224)

Describe what you did and why.

- create a single shared ownership form the citizens  and provider journey 
- add provider shared ownership routes

## Checklist

Before you ask people to review this PR:

- [x] Tests and rubocop should be passing: `bundle exec rake`
- [x] Github should not be reporting conflicts; you should have recently run `git rebase master`.
- [x] There should be no unneccessary whitespace changes. These make diffs harder to read and conflicts more likely. 
- [x] The PR description should say what you changed and why, with a link to the JIRA story.
- [x] You should have looked at the diff against master and ensured that nothing unexpected is included in your changes.
- [x] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: config/locales/en.yml [hunk 1]
```diff
@@ -372,7 +378,9 @@ en:
           title: Related content
           item_1: Legal aid
           item_2: More
-
+    shared_ownerships:
+      show:
+        heading_1: Does your client own thier home with anyone else?
```
Comment (mpw5) [inline config/locales/en.yml]: 'Thier' should be 'their'.

File: spec/requests/providers/shared_ownerships_spec.rb [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+      get providers_legal_aid_application_shared_ownership_path(legal_aid_application)
+    end
+
+    it 'returns http success' do
+      expect(response).to have_http_status(:ok)
+    end
+
+    it 'renders the shared ownership page' do
+      expect(response).to be_successful
+      expect(unescaped_response_body).to match('Does your client own thier home with anyone else?')
```
Comment (mpw5) [inline spec/requests/providers/shared_ownerships_spec.rb]: 'Thier' should be 'their'.

File: spec/requests/providers/shared_ownerships_spec.rb [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+      end
+    end
+
+    context 'with an invalid param' do
+      let(:params) { { legal_aid_application: { shared_ownership: '' } } }
+      it 're-renders the form with the validation errors' do
+        patch_request
+        expect(unescaped_response_body).to include('There is a problem')
+        expect(unescaped_response_body).to include('Select yes if you own your home with someone else')
+        expect(unescaped_response_body).to include('Does your client own thier home with anyone else?')
```
Comment (mpw5) [inline spec/requests/providers/shared_ownerships_spec.rb]: 'Thier' should be 'their'.

File: config/locales/en.yml [hunk 4]
```diff
@@ -34,6 +34,12 @@ en:
       date_input_fields:
         date_of_birth_label: 'Date of birth'
         date_of_birth_hint: 'For example, 31 3 1980'
+      shared_ownership_form:
+        shared_ownership_item:
+          partner_or_ex_partner:  Yes, a partner or ex-partner
+          housing_assocation_or_landlord: Yes, a housing association or landlord
+          friend_family_member_or_other_individual: Yes, a friend, family member or other individual
+          no_sole_owner: No, I am the sole owner
```
Comment (mpw5) [inline config/locales/en.yml]: This probably needs to be worded differently when a provider uses the screen. Something like 'No, my client is the sole owner'?

File: config/locales/en.yml (hunk 1 above)
Comment (tmckernan) [reply config/locales/en.yml]: fixed

File: spec/requests/providers/shared_ownerships_spec.rb (hunk 2 above)
Comment (tmckernan) [reply spec/requests/providers/shared_ownerships_spec.rb]: fixed

File: spec/requests/providers/shared_ownerships_spec.rb (hunk 3 above)
Comment (tmckernan) [reply spec/requests/providers/shared_ownerships_spec.rb]: fixed

File: config/locales/en.yml (hunk 4 above)
Comment (tmckernan) [reply config/locales/en.yml]: I have change this to "No, sole owner" to keep it generic

File: app/views/citizens/shared_ownerships/show.html.erb [hunk 5]
```diff
@@ -1,34 +1,6 @@
...
-        </div>
-      </div>
-
-      <div class="govuk-!-padding-bottom-6"></div>
-
-      <%= form.submit t('generic.continue'), id: 'continue', class: 'govuk-button' %>
-    <% end %>
-  </div>
-</div>
+<h1 class="govuk-heading-xl"><%= t('.heading_1') %></h1>
```
Comment (mpw5) [inline app/views/citizens/shared_ownerships/show.html.erb]: I think to get the layout on the page correct the partial needs to be in a two-thirds column inside a grid row. Something like:

```<div class="govuk-grid-row">
  <div class="govuk-grid-column-two-thirds">
    <!-- content -->
  </div>
</div>```

(The error summary shouldn't be in the two-thirds column, that can be full width.)

File: app/views/providers/shared_ownerships/show.html.erb [hunk 6]
```diff
@@ -0,0 +1,7 @@
+<% content_for :navigation do %>
+  <%= link_to t('generic.back'), '#', class: 'govuk-back-link', id: 'back' %>
+<% end %>
+<%= render partial: 'shared/forms/error_summary', locals: { model: @form } %>
+<h1 class="govuk-heading-xl"><%= t('.heading_1') %></h1>
```
Comment (mpw5) [inline app/views/providers/shared_ownerships/show.html.erb]: As above, the content should be in a two-thirds column.

**Pull request merged**